src/middlewares/auth/route.auth.ts

```diff
@@ -0,0 +1,28 @@
+import passport from 'passport';
+import { Request, Response, NextFunction } from 'express';
+import { StatusCodes } from 'http-status-codes';
+
+import CONSTANTS from '../../constants';
+const { UNAUTHORIZED, FORBIDDEN } = CONSTANTS.CONTROLLER_RESPONSE;
+
+const auth = (req: Request, res: Response, next: NextFunction) => {
+    passport.authenticate('jwt', { session: false }, (err, user, info) => {
+        if (err) next(err);
+
+        if (!req.headers['authorization']) {
+            return res
+                .status(StatusCodes.UNAUTHORIZED)
+                .json({ message: UNAUTHORIZED });
+        }
+
+        if (info?.message === 'invalid token') {
+            return res
+                .status(StatusCodes.FORBIDDEN)
+                .json({ message: FORBIDDEN });
+        }
+
+        next();
+    })(req, res);
```

Comment:
As far as I can see you're not passing "next" which usually breaks middleware flow.
So consider trying to refactor this auth approach a bit to separate the `passport.authenticate(...)` middleware and the authentication checking middleware.

Reply:
As far as I remember, `passport.authenticate(...)` adds a number of helper properties/methods to `req` object.
E.g. you can call `req.isAuthenticated(...)` in any subsequent middleware/handler to fin the auth result.
https://github.com/jwalton/passport-api-docs#passportauthenticatestrategyname-options-callback
Yeah, that's someone else's GH repo with better docs then Passport library provides XD
https://www.reddit.com/r/node/comments/8qdukp/why_is_passportsjs_s_docs_so_fucking_shitty/

Reply:
Since that's just a suggestion, approving the PR in order to not block the merge since that's Passport docs issues essentially.

Reply:
The link you provided is going straight into my bookmarks! I'm pleased I'm not the only one that feels the passport docs are inadequate (Spent the day fruitlessly googling). 

I'll try to implement the flow above - thanks for providing this :)

Reply:
Another roadblock :(. I could be wrong, but `req.isAuthenticated` seems to be used for local login strategies. I've tried calling `req.login` within the `passport.authentication` middleware as per the docs above, but it seems that false is always returned with `req.isAuthenticated`.  

For now, I've used `res.locals` to store the auth status in the passport middleware, and the authentication checking is handled separately. It's possible I'm wrong here, I've seen others have the same problem: https://stackoverflow.com/questions/59063175/passport-jwt-token-passed-user-found-but-isauthenticated-returns-false. I'm currently trying to finish up writing tests for the next task - so I'll take another crack at this in the next HT if there's a way around this.